addons/website_forum/populate/mail_message.py

```diff
@@ -0,0 +1,53 @@
+from random import choices
+from odoo import models
+from odoo.tools import populate
+
+CP_WEIGHTS = { '1': 35, '2': 30, '3': 25, '4': 10 }
+MAX_COMMENT = 4
+COMMENTS_RANGE = list(range(1,(MAX_COMMENT + 1)))
+WEIGHTS = list(CP_WEIGHTS.values())
+
+class Message(models.Model):
+    _inherit = "mail.message"
+    _populate_dependencies = ["forum.post"]
+    _populate_sizes = {'small': 100, 'medium': 1000, 'large': 90000 }
+
+    def _populate_factories(self):
+        return [
+            ("body", populate.constant("message_body_{counter}")),
+            ("message_type", populate.constant("comment")),
+            ("model", populate.constant("forum.post")),
+        ]
+
+    def _populate(self, size):
+        records = super()._populate(size)
+
+        post_ids = self.env.registry.populated_models['forum.post']
+        question_ids = self.env['forum.post'].search(
+            [('id', 'in', post_ids), ('parent_id', '=', None)]).mapped('id')
+        answer_ids = self.env['forum.post'].search(
+            [('id', 'in', post_ids), ('parent_id', '!=', None)]).mapped('id')
```

Comment:
This can also be simplified as follows:

```suggestion
        posts = self.env['forum.post'].browse(self.env.registry.populated_models['forum.post'])
        question_ids = posts.filtered(lambda p : not p.parent_id).ids
        answer_ids = (posts - questions).ids
```

If you aim at performance though, you might want to avoid fetching all records data to keep just the `id` with something like
```suggestion
        post_ids = self.env.registry.populated_models['forum.post']
        question_ids = self.env['forum.post'].search_read(
            [('id', 'in', post_ids), ('parent_id', '=', None)], [id])
        question_ids = self.env['forum.post'].search_read(
            [('id', 'in', post_ids), ('parent_id', '!=', False)], [id])
```